src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -282,7 +282,7 @@ private async ValueTask WriteHeadersAsync(HttpHeaders headers, string? cookiesFr
 
                         await WriteStringAsync(_headerValues[0], async, valueEncoding).ConfigureAwait(false);
 
-                        if (cookiesFromContainer != null && header.Key.KnownHeader == KnownHeaders.Cookie)
+                        if (cookiesFromContainer != null && header.Key.Equals(KnownHeaders.Cookie))
```

Comment:
We have the knownHeader from above, why not just do "knownHeader == KnownHeaders.Cookie"?

Reply:
`knownHeader` is no longer in scope after 65c59000fe2e54b3f21ea510115fec2275db6e81